File: preTateYoko.scala
import java.util._
object Main {
    val sc = new Scanner(System.in)
    val H = sc.nextInt
    val W = sc.nextInt
    val mat = Array.ofDim[String](H, W)
    // val arr = Array.fill(H)[String](sc.next)
    for (i <- 0 to H - 1){
        mat(i)(0) = sc.next //Array.fill(W)(sc.next)
    }
    val p = sc.nextInt
    val q = sc.nextInt

    for (i <- 0 to H - 1){
        mat(i) = mat(i)(0).split("")
    }

    def main(args: Array[String]): Unit = {
        var result = {
            if (mat(p)(q) == "#") -1
            else 0
        }

        for (i <- 0 to W - 1){
            if(mat(p)(i) == "#"){
                result += 1
            }
        }
        for (j <- 0 to H - 1){
            if(mat(j)(q) == "#"){
                result += 1
            }
        }
        println(result)
    }
}

File: preTateYoko.java
import java.util.*;
public class Main {
    static Scanner sc = new Scanner(System.in);
    static int H = sc.nextInt();
    static int W = sc.nextInt();
    static String[] arr = new String[H];
    static char[][] mat = new char[H][W];

  public static void main(String[] args){
    for(int i = 0; i < H; i++){
        arr[i] = sc.next();
    }

    for(int i = 0; i < H; i++){
        for(int j = 0; j < W; j++){
            mat[i][j] = arr[i].charAt(j);
        }
    }
    final int p = sc.nextInt();
    final int q = sc.nextInt();

    int result = 0;
    if(mat[p][q] == '#'){
        result = -1;
    }
    else {
        // result = 0;
    }
    

    for(int i = 0; i < H; i++){
        if(mat[i][q] == '#'){
            result += 1;
        }
        else {
            // 
        }
    }

    for(int j = 0; j < W; j++){
        if(mat[p][j] == '#'){
            result += 1;
        }
        else {
            // 
        }
    }

    System.out.println(result);

  }
}